src/ray/raylet/node_manager.cc

```diff
@@ -736,6 +741,73 @@ void NodeManager::DispatchTasks(
   local_queues_.MoveTasks(assigned_task_ids, TaskState::READY, TaskState::RUNNING);
 }
 
+std::pair<std::shared_ptr<Worker>, bool> NodeManager::GetWorker(
+    const WorkerID &worker_id) {
+  bool is_worker = true;
+  auto worker = worker_pool_.GetRegisteredWorker(worker_id);
+  if (!worker) {
+    is_worker = false;
+    worker = worker_pool_.GetRegisteredDriver(worker_id);
+    if (!worker) {
+      return std::make_pair(nullptr, true);
```

Comment:
Maybe simpler to use two functions `GetWorker` and `GetDriver` so that the semantics can be more clear?

Reply:
I think we need a function to get the pointer to the worker according to the worker id. We still need to check the return pointer to check whether it's a driver or worker if use two funcs.

Reply:
yes. An issue with original code is the second field `is_worker` in the returned pair is meaningless when the first field is nullptr.This is a minor issue though.